packages/sdk/src/services/encode.ts

```diff
@@ -0,0 +1,272 @@
+import { abi } from '../../abis/ExecutionModule.json';
+import { TickMath } from '@uniswap/v3-sdk';
+import { BigNumber, ethers } from 'ethers';
+import { closestTickAndFixedRate } from '../utils/math/tickHelpers';
+import { createAccountId, scaleAmount } from '../utils/helpers';
+import { CommandType, getCommand } from '../utils/routerCommands';
+import {
+  BaseTrade,
+  MakerTrade,
+  MethodParameters,
+  MultiAction,
+  SettleTradeMaker,
+  SettleTradeTaker,
+  TakerTrade,
+} from '../utils/types';
+import { getTokenInfo } from '../utils/constants';
+
+export async function encodeMakerOrder(
+  trade: MakerTrade,
+): Promise<MethodParameters> {
+  const multiAction = new MultiAction();
+
+  if (!trade.accountId) {
+    // open account
+    const accountId = await createAccountId(
+      trade,
+      true,
+      trade.fixedRateLower,
+      trade.fixedRateUpper,
+    );
+    trade.accountId = accountId;
+    encodeSingleOpenAccount(trade, multiAction);
+  }
+
+  let ethAmount = '0';
+  // todo: create a function (same code as maker order)
```

Comment:
perhaps fix the todos? this still feels like WIP

Reply:
fixed this todo [c36bf33](https://github.com/Voltz-Protocol/v2-off-chain-monorepo/pull/10/commits/c36bf330f7daf2dab92b10c2d0c1ef4b5e4e1cc0)
tho SDK v2 is still a WIP :)